src/gateway_config_miner.erl

```diff
@@ -0,0 +1,106 @@
+-module(gateway_config_miner).
+
+-behavior(gen_server).
+
+-include("gateway_config.hrl").
+-define(SERVICE, 'helium.local.api').
+
+%% gen_server
+-export([
+    start_link/0,
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%% api
+-export([
+    status/0,
+    pubkey/0,
+    add_gateway/3
+]).
+
+-record(state, {
+    connection = undefined :: grpc_client:connection() | undefined
+}).
+
+-spec status() -> {ok, [tuple()]} | {error, term()}.
+status() ->
+    gen_server:call(?MODULE, status).
+
+-spec pubkey() -> {ok, {PubKey :: binary(), OnboardingKey :: binary()}} | {error, term()}.
+pubkey() ->
+    gen_server:call(?MODULE, pubkey).
+
+-spec add_gateway(
+    Owner :: libp2p_crypto:pubkey_bin(),
+    Payer :: libp2p_crypto:pubkey_bin(),
+    Mode :: full | dataonly | light
+) -> {ok, binary()} | {error, term()}.
+add_gateway(Owner, Payer, Mode) ->
+    gen_server:call(?MODULE, {add_gateway, [Owner, Payer, Mode]}).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
```

Comment:
i added this missing function since you had referenced it as the start function in the gateway-config supervisor